### PR TITLE
fix(module:badge): fix number transition

### LINKED ISSUE #21674: [Badge] Weird rolling animation between 9 to 11
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/badge/#components-badge-demo-change](https://ant.design/components/badge/#components-badge-demo-change)

### Steps to reproduce
Change up and down between 9 to 11

### What is expected?
It should only roll a little

### What is actually happening?
Rolls one (or more) full cycle

| Environment | Info |
|---|---|
| antd | 4.0.0 |
| React | 16.13 |
| System | macOS 10.14.5 |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by wzhudev**
close #21674

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#21674 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix number transition error of the badge component  |
| 🇨🇳 Chinese |  修复 badge 组件数字动画错误  |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

屏幕录制2020-03-0321 12 59

### CODE REVIEW

File: components/badge/ScrollNumber.tsx [hunk 1]
```diff
@@ -82,14 +82,18 @@ class ScrollNumber extends React.Component<ScrollNumberProps, ScrollNumberState>
...
-      this.setState(
-        (__, props) => ({
-          animateStarted: false,
-          count: props.count,
-        }),
-        this.onAnimated,
-      );
+      // Let browser has time to reset the scroller before actually
+      // performing the transition.
+      setTimeout(() => {
```
Comment (afc163) [inline components/badge/ScrollNumber.tsx]: 用 requestAnimationFrame 可以么？

Comment (afc163) [reply components/badge/ScrollNumber.tsx]: 另外 setTimeout 需要在销毁时 clearTimeout 掉。

Comment (wendellhu95) [reply components/badge/ScrollNumber.tsx]: > 用 requestAnimationFrame 可以么？

试了一下不行。

Comment (wendellhu95) [reply components/badge/ScrollNumber.tsx]: 3.x 要支持 IE9 的话，不能用这个 API

File: components/badge/ScrollNumber.tsx [hunk 2]
```diff
@@ -82,17 +84,24 @@ class ScrollNumber extends React.Component<ScrollNumberProps, ScrollNumberState>
...
+            animateStarted: false,
+            count: props.count,
+          }),
+          this.onAnimated,
+        );
+      });
     }
   }
 
+  componentWillUnmount() {}
```
Comment (afc163) [inline components/badge/ScrollNumber.tsx]: 这里是空的？

**Comment by afc163** [comment]
测试覆盖率需要补上来。

**Pull request merged**